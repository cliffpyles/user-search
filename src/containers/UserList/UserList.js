import React from 'react'

export default ({ users }) => {
  return (
    <div>
      <h1>Users</h1>
      {users.map(user => {
        return <div key={user.id}>{user.login}</div>
      })}
    </div>
  )
}
