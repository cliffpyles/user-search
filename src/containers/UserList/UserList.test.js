import React from 'react'
import initializeStore from '../../store'
import UserList from './'

test('can render a list of users', () => {
  const store = initializeStore({
    search: {
      results: [
        { id: '1', login: 'seanlarkin' },
        { id: '2', login: 'mikebrocchi' },
        { id: '3', login: 'alexkureev' },
        { id: '4', login: 'balenterdi' },
        { id: '5', login: 'jameslong' }
      ]
    }
  })
  const result = render(<UserList store={store} />)

  expect(result).toMatchSnapshot()
})
