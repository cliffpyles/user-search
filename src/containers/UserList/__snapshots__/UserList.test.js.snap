// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a list of users 1`] = `
<div
  className="Page"
>
  <h1>
    Users
  </h1>
  <div
    className="Search"
  >
    <div
      className="TextField"
    >
      <label />
      <input
        onChange={[Function]}
        type="search"
      />
    </div>
    <button
      className="Button"
      onClick={[Function]}
      type="button"
    >
      search
    </button>
  </div>
  <div
    className="Card"
  >
    seanlarkin
  </div>
  <div
    className="Card"
  >
    mikebrocchi
  </div>
  <div
    className="Card"
  >
    alexkureev
  </div>
  <div
    className="Card"
  >
    balenterdi
  </div>
  <div
    className="Card"
  >
    jameslong
  </div>
</div>
`;
